packages/@okta/vuepress-site/docs/guides/sign-into-spa/main/index.md

```diff
@@ -0,0 +1,185 @@
+
+<StackSelector snippet="config" noSelector />
+
+## Add a button to sign in
+
+In your application, you want to provide a button to sign the user in. When the user clicks this button, they are redirected to the Okta-hosted sign-in page so they can authenticate.
+
+> **Note:** To customize this sign-in page, see the [Style the Widget](/docs/guides/style-the-widget/style-okta-hosted/).
+
+<StackSelector snippet="login-redirect" noSelector />
+
+## Handle the callback from Okta
+
+After Okta authenticates a user, they're redirected back to your application via the callback route you [Define a callback route](#define-a-callback-route). When Okta redirects back, the URL fragment (the portion after `#`) contains either tokens for the user or an error if something went wrong.
+
+Your application must parse this information, and if tokens are present, store the user's tokens. Our SDK does this for you.
+
+<StackSelector snippet="handle-callback" noSelector />
+
+## Require authentication
+
+In many applications, you want to prevent the user from accessing certain routes or sections unless they are signed in. You can require authentication across the entire app or just require it for particular routes or controllers. Any route that doesn't specifically require authentication is accessible without signing in (also called anonymous access).
+
+It's important to note that protecting routes in your SPA does not truly prevent the user from accessing those parts of your application. After all, it's JavaScript running in the browser and anyone could open the browser's developer tools and change things! Protecting routes provides a consistent and good experience for your users. The real security enforcement must be done in the API that your single-page app calls (see [Use the access token](#use-the-access-token)). See also the [Protect your API endpoints guide](/docs/guides/protect-your-api/).
+
+### Require authentication for a specific route
+
+If you want the user to only have access to a route if they are signed in, require authentication for just those routes.
+
+<StackSelector snippet="reqauthspecific" noSelector />
+
+### Require authentication for everything
+
+For some applications, you may want to require the user to be authenticated for all routes.
+
+<StackSelector snippet="reqautheverything" noSelector />
+
+## Get info about the user
+
+When a user signs in, their profile information (stored in Okta) is made available to your application. It's common to use this info to update your app's UI.
+
+By default, the profile items (called "claims") that Okta returns include the user's email address, first name, and last name. 
+
+<StackSelector snippet="getuserinfo" noSelector />
+
+<!-- You can also customize the items (called claims) that are returned from Okta. See [Token customization guide]. -->
```

Comment:
why is this commented out? there are numerous customize token guides, starting with this one: https://developer.okta.com/docs/guides/customize-tokens-returned-from-okta/overview/

Reply:
I really have no idea ;-) 

But I've uncommented it now, and added in a proper link.